nestkernel/eprop_archiving_node.h

```diff
@@ -117,21 +199,99 @@ class EpropArchivingNodeRecurrent : public EpropArchivingNode< HistEntryEpropRec
   //! Copy constructor.
   EpropArchivingNodeRecurrent( const EpropArchivingNodeRecurrent& );
 
-  //! Create an entry in the eprop history for the given time step and surrogate gradient.
+  //! Select the surrogate gradient function.
+  double ( EpropArchivingNodeRecurrent::*select_surrogate_gradient(
+    std::string surrogate_gradient_function ) )( double, double, double, double, double, double )
+  {
+    const std::map< std::string,
+      double ( EpropArchivingNodeRecurrent::* )( double, double, double, double, double, double ) >
+      surrogate_gradient_funcs = { { "piecewise_linear",
+                                     &EpropArchivingNodeRecurrent::compute_piecewise_linear_surrogate_gradient },
+        { "exponential", &EpropArchivingNodeRecurrent::compute_exponential_surrogate_gradient },
+        { "fast_sigmoid_derivative", &EpropArchivingNodeRecurrent::compute_fast_sigmoid_derivative_surrogate_gradient },
+        { "arctan", &EpropArchivingNodeRecurrent::compute_arctan_surrogate_gradient } };
+
+    auto found_entry_it = surrogate_gradient_funcs.find( surrogate_gradient_function );
+    if ( found_entry_it == surrogate_gradient_funcs.end() )
+    {
+      std::string error_message = "Surrogate gradient / pseudo-derivate function surrogate_gradient_function from [";
+      for ( const auto& surrogate_gradient_func : surrogate_gradient_funcs )
+      {
+        error_message += " \"" + surrogate_gradient_func.first + "\",";
+      }
+      error_message.pop_back();
+      error_message += " ] required.";
+
+      throw BadProperty( error_message );
+    }
+    else
+    {
+      return found_entry_it->second;
+    }
+  }
+
+  //! Compute the surrogate gradient with a piecewise linear function around the spike time (used, e.g., in Bellec et
+  //! al., 2020).
+  double compute_piecewise_linear_surrogate_gradient( const double r,
+    const double v_m,
+    const double v_th_adapt,
+    const double V_th,
```

Comment:
Why does is the V, not v?

Reply:
This follows the NEST legacy uppercase style for some specific constant parameter variables (`V_th`, `C_m`, `E_L` etc.). We use lowercase for dynamic variables (like `v_th_adapt`, `v_m`, etc.). In any case, I will now prepare a longer planned PR to remove the passing of the `V_th` altogether.

Reply:
Fixed in https://github.com/nest/nest-simulator/pull/3207/commits/1ffc7520d68eb4112ba1afd08bf187020e0c629f